### DISCUSSION TITLE
v7 - onHosts?

### DISCUSSION THREAD

**Discussion by leevigraham** [deployphp/deployer#2470] [Help needed]
Was `onHosts` removed in v7? Is there any other way to restrict a task to certain hosts?

**Comment by antonmedv** [comment]
Yes, just use select() with a selector.

**Reply by leevigraham** [reply]
What's an example of the selector string? Is it just comma delimited hostnames?

**Reply by antonmedv** [reply]
Yes, and label=value

**Reply by antonmedv** [reply]
prod.com,role=db

**Reply by mbrodala** [reply]
The necessary change:

```diff
-$task->onHosts('<hostname1>', '<hostname2>');
+$task->select('<hostname1>, <hostname2>');
```

**Comment by russellfeeed** [comment]
can you select all hosts?

something like...
```
task('mytask', ....) ->select('*')
```

to elaborate, I'm dynamically creating the hosts (I pull them from an API). Then I want the user to select which host they want to run the task on from the list all all known hosts. I think in v6 that worked if you just used `->select()`. But in V7 it doesn't seem to work the same (I get a number of parameters error)

**Reply by mbrodala** [reply]
Well, you can just skip `select()` altogether. Tasks by default apply to all hosts.

**Reply by antonmedv** [reply]
Also you can specify “all” if you really want to.
